### DISCUSSION TITLE
Should the node adapter have a feature to exit after a certain amount of idle time?

### DISCUSSION THREAD

**Discussion by jyasskin** [withastro/roadmap#438] [Proposal]
I'd like to have my server exit cleanly if it hasn't processed a request for a reasonable amount of time, in order to cooperate with [fly.io's scale-to-zero support](https://fly.io/blog/fly-machines/#stopping-and-scaling-to-zero). I could do this by writing my own little server and using the node adapter's `middleware` mode, but if this is something other people also want, it seems like it'd be straightforward to add an option to enable it in the `standalone` mode.

Is it worth sending you a PR for this? I'd probably add a `maxIdleSeconds: number` field to the [`UserOptions`](https://github.com/withastro/astro/blob/main/packages/integrations/node/src/types.ts) type unless you prefer something else.

**Comment by florian-lefebvre** [comment]
I don't think we want to support this in core so using middleware mode seems to be the best solution here!
